nuxeo-drive-rest-api/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/drive/NuxeoDriveObject.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Antoine Taillefer <ataillefer@nuxeo.com>
+ */
+package org.nuxeo.ecm.restapi.server.jaxrs.drive;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.nuxeo.common.Environment;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.exceptions.WebResourceNotFoundException;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+
+/**
+ * @since 9.10
+ */
+@WebObject(type = "drive")
+@Produces(MediaType.APPLICATION_JSON)
+public class NuxeoDriveObject extends DefaultObject {
+
+    public static final String NUXEO_DRIVE_CONFIGURATION_FILE = "nuxeo-drive-config.json";
+
+    /**
+     * Retrieves the Nuxeo Drive global configuration.
+     *
+     * @implNote The configuration file is expected to be in the server's configuration folder, copied from the
+     *           {@code drive} template.
+     */
+    @GET
+    @Path("configuration")
+    public String getConfiguration() {
+        File configurationFolder = Environment.getDefault().getConfig();
+        File configurationFile = new File(configurationFolder, NUXEO_DRIVE_CONFIGURATION_FILE);
+        if (!configurationFile.exists()) {
+            throw new WebResourceNotFoundException("Nuxeo Drive configuration file not found.");
+        }
+
+        try {
+            byte[] bytes = Files.readAllBytes(Paths.get(configurationFile.getPath()));
+            return new String(bytes, StandardCharsets.UTF_8);
```

Comment:
It's probably better to let JAX-RS stream the file with something like (untested) `return Response.ok().entity(configurationFile).type("application/json").build()`

Reply:
This is not up-to-date, I will backport the latest changes made on the master PR: https://github.com/nuxeo/nuxeo-drive-server/pull/59/commits/7cb2acfdfa2a216ce22ceb194175b0b65cabf9ea#diff-3920b1a536d11cd4b666e8e01232545fR66

Reply:
`Response.ok().entity(configurationFile).type(MediaType.APPLICATION_JSON).build()` works and is indeed better since we don't unnecessarily load the file content in a string.